src/Features/CSharp/Portable/QuickInfo/OnTheFlyDocsUtilities.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.QuickInfo;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.QuickInfo;
+
+internal static class OnTheFlyDocsUtilities
+{
+    public static ImmutableArray<OnTheFlyDocsRelevantFileInfo?> GetAdditionalOnTheFlyDocsContext(Solution solution, ISymbol symbol)
+    {
+        var parameters = symbol.GetParameters();
+        var typeArguments = symbol.GetTypeArguments();
+        var parameterStrings = parameters.Select(parameter =>
+        {
+            var typeSymbol = parameter.Type;
+            var typeSyntaxReference = typeSymbol.DeclaringSyntaxReferences.FirstOrDefault();
+            if (typeSyntaxReference is not null)
+            {
+                var typeSpan = typeSyntaxReference.Span;
+                var syntaxReferenceDocument = solution.GetDocument(typeSyntaxReference.SyntaxTree);
+                if (syntaxReferenceDocument is not null)
+                {
+                    return new OnTheFlyDocsRelevantFileInfo(syntaxReferenceDocument, typeSpan);
+                }
+            }
+
+            return null;
+
+        }).ToImmutableArray();
+
+        var typeArgumentStrings = typeArguments.Select(typeArgument =>
```

Comment:
both `parameters` and `typeArguments` are arrays of types from `ISymbol`, so you should be able to concat them all into one array and iterate together.

Either that or extract the `Select` implementation into a helper that takes the `ISymbol` and creates the `OnTheFlyDocsRelevantFileInfo` and use for both (instead of duplicate implementation)